shoop/core/models/product_shops.py

```diff
@@ -103,7 +103,7 @@ def get_visibility_errors(self, customer):
                 _('The Product is invisible to users not logged in.'),
                 code="product_not_visible_to_anonymous")
 
-        if self.visibility_limit == ProductVisibility.VISIBLE_TO_GROUPS:
+        if customer and self.visibility_limit == ProductVisibility.VISIBLE_TO_GROUPS:
```

Comment:
Hmm... This should probably use `is_logged_in`, because `AnonymousUser`s won't have `groups` either.

Also, can you add a test case for this fix?